src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -111,10 +111,12 @@ static List<Type> GetHandlerMessageTypes(List<Type> list)
                     {
                         var messageType = @interface.GetGenericArguments()[0];
                         foundMessageTypes.Add(messageType);
+
+                        foundMessageTypes.AddRange(messageType.Assembly.GetTypes().Where(t=> messageType.IsAssignableFrom(t)));
```

Comment:
I like it, we would also only need the second pass if we find a message that isn't assignable to `IMessage` so users with the default conventions wouldn't be affected